src/jqLite.js

```diff
@@ -420,13 +420,15 @@ function jqLiteHasClass(element, selector) {
 
 function jqLiteRemoveClass(element, cssClasses) {
   if (cssClasses && element.setAttribute) {
+    var existingClasses = (' ' + (element.getAttribute('class') || '') + ' ')
+                            .replace(/[\n\t]/g, ' ');
+
     forEach(cssClasses.split(' '), function(cssClass) {
-      element.setAttribute('class', trim(
-          (' ' + (element.getAttribute('class') || '') + ' ')
-          .replace(/[\n\t]/g, ' ')
-          .replace(' ' + trim(cssClass) + ' ', ' '))
-      );
+      cssClass = trim(cssClass);
+      existingClasses = existingClasses.replace(' ' + cssClass + ' ', ' ');
     });
+
+    element.setAttribute('class', trim(existingClasses));
```

Comment:
Should you do a check to ensure that we don't set this unnecessarily which potentially causes unnecessary renders?

Ref: https://github.com/jquery/jquery/blob/master/src/attributes/classes.js#L41-L45